hotshot-task-impls/src/consensus/handlers.rs

```diff
@@ -315,11 +314,11 @@ pub async fn send_high_qc<TYPES: NodeType, V: Versions, I: NodeImplementation<TY
         if is_epoch_root {
             // For epoch root QC, we are sending high QC and state cert
             let Some(state_cert) = state_cert else {
-                bail!("We are sending an epoch root QC but we don't have the current state cert.");
+                bail!("We are sending an epoch root QC but we don't have the corresponding state cert.");
             };
             ensure!(
-                Some(state_cert.epoch) == cur_epoch,
-                "We are sending an epoch root QC but we don't have the current state cert."
+                Some(state_cert.epoch) == high_qc.epoch(),
```

Comment:
Can you also add a checks that the state_cert.data.leaf_commits are the same

Reply:
We can't since it only contains epoch_number, view_number, block_height and block merkle root